mkcgo: Add missing error symbols to {load,unload}_init_3

sergiodj opened the pull request:
When `initForCheckVersion` is called, and if there is an error in openssl's initialization, the codepath can reach `mkcgo_err_retrieve` which may invoke `_mkcgo_ERR_get_error_all`.  This will make the program segfault because the symbol for `ERR_get_error_all` is NULL.

We need to load a couple more symbols from openssl in order to properly handle the error and display a sane message to the user.

qmuntal commented on internal/ossl/zossl.c:
```diff
@@ -633,6 +633,8 @@ void __mkcgo_load_init_3(void* handle) {
 	__mkcgo__dlsym(EVP_MD_free)
 	__mkcgo__dlsym(EVP_MD_get0_provider)
 	__mkcgo__dlsym(EVP_default_properties_is_fips_enabled)
+	__mkcgo__dlsym(ERR_get_error_all)
```
`zossl.c` is autogenerated, you should update `shims.h` instead and then run `go generate ./...`.

You should update the functions definitions in this way:

```c
void ERR_error_string_n(unsigned long e, char *buf, size_t len) __attribute__((tag(""),tag("init_3"),noerror));
unsigned long ERR_get_error_all(const char **file, int *line, const char **func, const char **data, int *flags) __attribute__((tag("3"),tag("init_3"),noerror));
```

sergiodj replied:
Ah, gotcha.  Thanks.  Let me update the PR and force push.

sergiodj replied:
Done.

sergiodj committed:
mkcgo: Add missing error symbols to {load,unload}_init_3

When `initForCheckVersion` is called, and if there is an error in
openssl's initialization, the codepath can reach `mkcgo_err_retrieve`
which may invoke `_mkcgo_ERR_get_error_all`.  This will make the
program segfault because the symbol for `ERR_get_error_all` is NULL.

We need to load a couple more symbols from openssl in order to
properly handle the error and display a sane message to the user.

Signed-off-by: Sergio Durigan Junior <sergiodj@chainguard.dev>

qmuntal commented:
Thanks!
